File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int l = 0;
        int r = 0;
        

        while (r < nums.size()) {
            if (nums[r] != val) {
                nums[l] = nums[r];
                l++;
            }
            r++;
        }
        return l;
    }
};

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        sets = {}

        for i, val in enumerate(nums):
            if val in sets and abs(sets[val] - i) <= k:
                return True
            else:
                sets[val] = i

        return False
 
